frontend/user/configProfile/configProfileController.js

```diff
@@ -1,37 +1,51 @@
 'use strict';
 
 (function () {
-    var app = angular.module("app");
+    const app = angular.module("app");
 
-    app.controller("ConfigProfileController", function ConfigProfileController($state, STATES,
-        CropImageService, AuthService, UserService, ImageService, $rootScope, $q, MessageService, $mdDialog, ObserverRecorderService) {
+    app.controller("ConfigProfileController", function ConfigProfileController($state, STATES, $stateParams, ProfileService,
+        CropImageService, AuthService, UserService, ImageService, $rootScope, SCREEN_SIZES, MessageService, $mdDialog, ObserverRecorderService) {
 
-        var configProfileCtrl = this;
+        const configProfileCtrl = this;
 
         // Variable used to observe the changes on the user model.
-        var observer;
-
-        configProfileCtrl.user = AuthService.getCurrentUser();
-        configProfileCtrl.newUser = _.cloneDeep(configProfileCtrl.user);
-        configProfileCtrl.loading = false;
+        let observer;
         configProfileCtrl.cpfRegex = /^\d{3}\.\d{3}\.\d{3}\-\d{2}$/;
-        configProfileCtrl.photo_url = configProfileCtrl.newUser.photo_url;
         configProfileCtrl.loadingSubmission = false;
 
-        var HAS_ONLY_ONE_INSTITUTION_MSG = "Esta é a única instituição ao qual você é vinculado." +
-            " Ao remover o vínculo você não poderá mais acessar o sistema," +
-            " exceto por meio de novo convite. Deseja remover?";
+        const DELETE_ACCOUNT_ALERT = "Ao excluir sua conta você não poderá mais acessar o sistema," +
+            "exceto por meio de novo convite. Deseja realmente excluir sua conta?";
 
-        var HAS_MORE_THAN_ONE_INSTITUTION_MSG = "Ao remover o vínculo com esta instituição," +
-            " você deixará de ser membro" +
-            " e não poderá mais publicar na mesma," +
-            " no entanto seus posts existentes serão mantidos. Deseja remover?";
+        configProfileCtrl.$onInit = () => {
+            configProfileCtrl._setupUser();
+        }
 
-        var DELETE_ACCOUNT_ALERT = "Ao excluir sua conta você não poderá mais acessar o sistema," +
-            "exceto por meio de novo convite. Deseja realmente excluir sua conta?";
+        configProfileCtrl._setupUser = () => {
+            if(configProfileCtrl.canEdit()) {
+                configProfileCtrl.user = AuthService.getCurrentUser();
+                configProfileCtrl.newUser = _.cloneDeep(configProfileCtrl.user);
+                observer = ObserverRecorderService.register(configProfileCtrl.user);
+                configProfileCtrl._checkUserName();
+            } else {
+                UserService.getUser($stateParams.userKey)
+                    .then(user => configProfileCtrl.user = user);
+            }
+        }
+        
+        configProfileCtrl._checkUserName = () => {
```

Comment:
I changed it to just set to replace the name 'Unknown' to an empty string because the user does not need to know that, when he is new in the application, his default name is set to "Unknown". So in the edit profile form, he just sees that his name is empty.